document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresIdentifierAccessingExpressionVisitor.java

```diff
@@ -0,0 +1,37 @@
+package org.hypertrace.core.documentstore.postgres.query.v1.vistors;
+
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.DOC_PATH_SEPARATOR;
+import static org.hypertrace.core.documentstore.postgres.utils.PostgresUtils.JSON_FIELD_ACCESSOR;
+
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.postgres.query.v1.PostgresQueryParser;
+
+public class PostgresIdentifierAccessingExpressionVisitor
+    extends PostgresSelectTypeExpressionVisitor {
+  private final String baseField;
+
+  public PostgresIdentifierAccessingExpressionVisitor(
+      final PostgresSelectTypeExpressionVisitor baseVisitor, final String baseField) {
+    super(baseVisitor);
+    this.baseField = baseField;
+  }
+
+  @Override
+  public PostgresQueryParser getPostgresQueryParser() {
+    return baseVisitor.getPostgresQueryParser();
```

Comment:
Curious: In the `PostgresIdentifierReplacingExpressionVisitor` class, we initially verify the existence of the `postgresQueryParser`. Afterward, we reference the `baseVisitor` as follows:

```java
postgresQueryParser != null ? postgresQueryParser : baseVisitor.getPostgresQueryParser()
```

Why aren't we doing the same here and directly accessing it from the `baseVisitor`?

Reply:
Actually, that is also not required as we are not passing in any `baseVisitor` in the constructor. Updated there to get rid of the unnecessary conditional statement.